### PR TITLE
Replace unnecessary imports with forward declarations

### PR DESCRIPTION
**Pull request by sigito**
Partially addresses #1523.

### CODE REVIEW

**Comment by willlarche** [comment]
Running extra tests internally. Then it will be unblocked.

File: components/ActivityIndicator/examples/ActivityIndicatorExample.m
Comment (randallli) [inline components/ActivityIndicator/examples/ActivityIndicatorExample.m:21]: nice! the supplementa/ is required for bazel build.

File: components/FeatureHighlight/src/private/MDCFeatureHighlightView+Private.h
Comment (randallli) [inline components/FeatureHighlight/src/private/MDCFeatureHighlightView+Private.h:19]: I think this is an incorrect transform.

File: components/FlexibleHeader/examples/supplemental/FlexibleHeaderWrappedSupplemental.m
Comment (randallli) [inline components/FlexibleHeader/examples/supplemental/FlexibleHeaderWrappedSupplemental.m:24]: interesting. how did you detect that the component wasn't being used?

File: components/Ink/examples/InkTypicalUse.m
Comment (randallli) [inline components/Ink/examples/InkTypicalUse.m]: If possible, I think we want to preserve the space between the import categories.

**Review by randallli — APPROVED** [review]
LGTM As long as this passes test.

**Comment by randallli** [comment]
testing it in cl/175877403.

File: components/FeatureHighlight/src/private/MDCFeatureHighlightView+Private.h
Comment (sigito) [reply components/FeatureHighlight/src/private/MDCFeatureHighlightView+Private.h:19]: I would disagree for a few reasons:
1. Since this private header extends the API of the `MDCFeatureHighlightView`, it should include its header directly.
1. `MaterialFeatureHighlight` is own component header. Components should not include own headers.

File: components/FlexibleHeader/examples/supplemental/FlexibleHeaderWrappedSupplemental.m
Comment (sigito) [reply components/FlexibleHeader/examples/supplemental/FlexibleHeaderWrappedSupplemental.m:24]: I have no idea. It's been so long ago.

File: components/ActivityIndicator/examples/supplemental/ActivityIndicatorExampleSupplemental.m
Comment (randallli) [inline components/ActivityIndicator/examples/supplemental/ActivityIndicatorExampleSupplemental.m:22]: we are using nsstring so we should keep this.

File: components/AnimationTiming/examples/supplemental/AnimationTimingExampleSupplemental.m
Comment (randallli) [inline components/AnimationTiming/examples/supplemental/AnimationTimingExampleSupplemental.m:17]: still need this.
in fact we need CoreGraphics too since we are using CGSize CGFloat et al

File: components/Buttons/examples/supplemental/ButtonsTypicalUseSupplemental.m
Comment (randallli) [inline components/Buttons/examples/supplemental/ButtonsTypicalUseSupplemental.m:22]: lets double check all of these.

**Review by randallli — CHANGES_REQUESTED** [review]
Can we sync with develop since there are some tests failing that have been fixed there.

**Review by randallli — APPROVED** [review]

File: components/FeatureHighlight/src/private/MDCFeatureHighlightView+Private.h
Comment (ianegordon) [reply components/FeatureHighlight/src/private/MDCFeatureHighlightView+Private.h:19]: I agree with @sigito , but happy to discuss if you feel strongly.

**Comment by ianegordon** [comment]
Waiting for a +1 from kokoro.

**Comment by ianegordon** [comment]
Tested locally.  Archiving and UITests on the sim.
kokoro is in flux, so I'm going to land.
@sigito Huge thanks for all this cleanup and your patience!
🎆 👍

**Comment by randallli** [comment]
tests pass for me too.

**Pull request merged**
